index.js

```diff
@@ -672,7 +667,24 @@ function fastifyView (fastify, opts, next) {
       }
       this.header('Content-Type', 'text/html; charset=' + charset)
       this.send(html)
-    })
+    }
+
+    data = Object.assign({}, defaultCtx, this.locals, data)
+    // Append view extension (Eta will append '.eta' by default,
+    // but this also allows custom extensions)
+    page = getPage(page, 'eta')
+    if (opts?.async ?? globalOptions.async) {
+      engine
+        .renderFile(page, data, config)
+        .then(sendContent)
+        .catch(err => this.send(err))
+    } else {
+      engine.renderFile(page, data, config, (err, html) => {
+        err
+          ? this.send(err)
+          : sendContent(html)
+      })
+    }
```

Comment:
I don't think we should have this if. We should not require the user to specify if `renderFile` is returning a promise or not.

Reply:
Can you explain why it's needed?

Reply:
Because i do not know any other way to fix the issue. Also this `if` is done in the `viewEjs` function .  

So if you have a better way for the fix i would appreciate any other idea as I hove none.

Reply:
I have no time to look into it atm.